terraform/install.sh

```diff
@@ -256,28 +258,30 @@ getExternalIp() {
 # Install Load Generator service and start generating synthetic traffic to Sandbox
 loadGen() {
   log "Running load generator"
-  # remove existing load generator
-  old_loadgen=$(gcloud compute instances list --project $project_id --filter="name:loadgenerator*" --format="value(name)")
-  if [[ -n "$old_loadgen" ]]; then
-    loadgen_zone=$(gcloud compute instances list --project $project_id --filter="name:loadgenerator*" --format="value(zone)")
-    gcloud compute instances delete $old_loadgen --zone $loadgen_zone --quiet
-  fi
   # launch a new load generator
-  ../loadgenerator/loadgen autostart $external_ip
+  pushd loadgen/
+  terraform init -lock=false
+  terraform apply --auto-approve -var="project_id=${project_id}" -var="external_ip=${external_ip}"
+  popd
+
   # find the IP of the load generator web interface
   TRIES=0
-  while [[ $(curl -sL -w "%{http_code}"  "http://$loadgen_ip:8080" -o /dev/null --max-time 1) -ne 200  && \
+  while [[ $(curl -sL -w "%{http_code}"  "http://$loadgen_ip:8089" -o /dev/null --max-time 1) -ne 200  && \
       "${TRIES}" -lt 20  ]]; do
     log "waiting for load generator instance..."
-    sleep 1
-    loadgen_ip=$(gcloud compute instances list --project "$project_id" \
-                                               --filter="name:loadgenerator*" \
-                                               --format="value(networkInterfaces[0].accessConfigs.natIP)")
+    sleep 10
+    loadgen_ip=$(kubectl get service locust-master -o jsonpath='{.status.loadBalancer.ingress[0].ip}');
+     [ -z "$loadgen_ip" ] && sleep 10;
     TRIES=$((TRIES + 1))
   done
-  if [[ $(curl -sL -w "%{http_code}"  "http://$loadgen_ip:8080" -o /dev/null  --max-time 1) -ne 200 ]]; then
+  if [[ $(curl -sL -w "%{http_code}"  "http://$loadgen_ip:8089" -o /dev/null  --max-time 1) -ne 200 ]]; then
     log "error: load generator unreachable"
   fi
+
+  # Make kubectx alias for this kubectl context
+  kubectx loadgenerator=.
+  # Return kubectl context to the main cluster and show this to the user
+  log $(kubectx main)
```

Comment:
These should be removed

Reply:
i disagree, I think it's best to return the kubectl context to the main one because it's what a user would be more likely to expect. they shouldn't need to really look at what's going on with the load generator.

Reply:
Yeah, I also think we should return the user to the main cluster after provisioning the load generator

I'm not so sure about putting it in a log statement though, it would probably be better to do it as two lines to separate the logging from the functionality

Reply:
Good point Gloria, agree - let's keep it then.

Reply:
yeah - i think i'd run `kubectx main` and then log `kubectx` to print the current context?